modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLayout from '@clayui/layout';
+import React from 'react';
+
+interface IItem {
+	label: string;
+	symbol: string;
+}
+
+interface ITranslations {
+	[key: string]: string;
+}
+
+export type ButtonSize = null | 'regular' | 'small';
+
+export type Variant =
+	| null
+	| 'default'
+	| 'language selector'
+	| 'translation selector';
+
+export interface IProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
+
+	/**
+	 * Labels for the aria attributes
+	 */
+	ariaLabels?: {
+		default: string;
+		translated: string;
+		untranslated: string;
+		manageTranslations: string;
+	};
+
+	/**
+	 * Button size to use "regular" or "default"
+	 */
+	buttonSize?: ButtonSize;
+
+	/**
+	 * List of locales to allow localization for
+	 */
+	locales: Array<IItem>;
+
+	/**
+	 * Callback that gets called the "Manage Translations" drop down item is clicked
+	 */
+	onManageTranslations?: () => void;
+
+	/**
+	 * Callback that gets called when a selected locale gets changed
+	 */
+	onSelectedLocaleChange?: (val: IItem) => void;
+
+	/**
+	 * Exposes the currently selected locale
+	 */
+	selectedLocale?: IItem;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap?: string;
+
+	/**
+	 * Translations provided to the component to be used and modified by it
+	 */
+	translations: ITranslations;
+
+	/**
+	 */
+	variant?: Variant;
+}
+
+const TranslationSelector: React.FunctionComponent<IProps> = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('not-translated'),
+	},
+	buttonSize = 'regular',
+	locales,
+	onManageTranslations = () => {},
+	onSelectedLocaleChange = () => {},
+	selectedLocale,
+	spritemap,
+	translations,
+	variant = 'default',
+}: IProps) => {
+	const [active, setActive] = React.useState(false);
+
+	const defaultLocale = locales[0];
```

Comment:
Maybe it wouldn't be better to leave this configurable? in this case we are always relying on the default value to always be first in the list. And in this case, whoever passes the data must always organize list before rendering.

I think this is also a criterion for Translation, right? always list the default language as first?

Reply:
Yes sure, this was me being lazy, we can totally make it configurable - let me fix that.

Reply:
Yes you're right, the default language should be the first one we "render" in that list.

Reply:
Updated in [`84fb294`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/84fb294ca0bf424a10c89db94cfa4a2fab0b00f5)

Reply:
Thanks for checking @julien